purchases/src/main/kotlin/com/revenuecat/purchases/PurchasesFactory.kt

```diff
@@ -127,6 +139,25 @@ internal class PurchasesFactory(
         return checkCallingOrSelfPermission(permission) == PackageManager.PERMISSION_GRANTED
     }
 
+    private fun createDiagnosticsManagerIfNeeded(
+        context: Context,
+        backend: Backend,
+        dispatcher: Dispatcher,
+        diagnosticsEnabled: Boolean
+    ): DiagnosticsManager? {
+        return if (diagnosticsEnabled) {
+            DiagnosticsManager(
+                DiagnosticsFileHelper(FileHelper(context)),
+                DiagnosticsAnonymizer(),
```

Comment:
This will cause a conflict when merging with #782 but we can easily solve it then.